service/controller/resource/app/desired.go

```diff
@@ -203,17 +205,48 @@ func (r *Resource) newApp(clusterConfig v1alpha1.ClusterGuestConfig, appSpec key
 	}
 }
 
-func (r *Resource) newAppSpecs() []key.AppSpec {
-	switch r.provider {
-	case "aws":
-		return append(key.CommonAppSpecs(), awskey.AppSpecs()...)
-	case "azure":
-		return append(key.CommonAppSpecs(), azurekey.AppSpecs()...)
-	case "kvm":
-		return append(key.CommonAppSpecs(), kvmkey.AppSpecs()...)
-	default:
-		return key.CommonAppSpecs()
+func (r *Resource) newAppSpecs(ctx context.Context, cr v1alpha1.ClusterGuestConfig) ([]key.AppSpec, error) {
+	req := searcher.Request{
+		ReleaseVersion: cr.ReleaseVersion,
+	}
+
+	res, err := r.clusterClient.Release.Searcher.Search(ctx, req)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	if len(res.Apps) == 0 {
+		return nil, microerror.Maskf(executionFailedError, "no apps in release %#q", req.ReleaseVersion)
```

Comment:
Do we have apps in all releases?

Reply:
No, but this cluster-operator would run with specific release only, which would include apps.